### DISCUSSION TITLE
Usage statistics for enterprise users

### DISCUSSION THREAD

**Discussion by benny-yamagata** [community/community#70024] [Enterprise]
### Select Topic Area

Question

### Body

Will there be some sort of dashboard where I can see things like usage statistics?

I want to know how many users are using copilot per day, how many lines of code generated, prompts entered, breakdowns by user, usage cases, etc. 

I would really like to dig in and see how my developers are using the product, what for, and how much is it really doing for us compared to what they are doing on their own.  

Also, would love to see a section there for code that was possibly pulled from public repositories, as well as the licenses attributed. 

Is this something that is coming soon?  Or any plans in the future?

**Comment by Akash1134** [comment]
Hi @benny-yamagata ,

To answer your needs for usage statistics partially, we released our beta version of the [User Management API](https://docs.github.com/en/rest/copilot/copilot-for-business?apiVersion=2022-11-28) for GitHub Copilot for Business sometime back. It provides the below statistics:

![Screenshot 2023-10-16 at 6 16 15 PM](https://github.com/community/community/assets/95834130/7b0b1125-8c82-4974-a5ab-f1dbc7b0ac97)

>> Note: You must authenticate using an access token with the manage_billing:copilot scope and be an admin on the Copilot for Business organization you're updating to use this endpoint.

You can also manage your Copilot for Business subscription (like reviewing audit logs) in this documentation: [Manage Copilot for Business](https://docs.github.com/en/copilot/managing-copilot-for-business).

We'd love for you to try it out and let us know what you think!

**Reply by wangliguang111** [reply]
This is not a statistical function, this is a business account operating function. And the stats feature is what everyone wants now

**Reply by benny-yamagata** [reply]
as @wangliguang111 has suggested, this is not a statistics or usage function.  This is administration of copilot for an enterprise org. 

@Akash1134 , is there a future work item to retrieve these types of statistics?  Many organizations are asking about ROI and how helpful copilot is actually to their workers, and this could really help inform those questions.    Also, sorry for the wayyyyyyy late reply, our company was reorg right after I posted this and it has been hectic for months.  You know how it goes.
